Sources/Datadog/DatadogCore/DatadogCore.swift

```diff
@@ -158,11 +158,46 @@ extension DatadogCore: DatadogV1CoreProtocol {
         return v1Features[key] as? T
     }
 
+    func scope<T>(for featureType: T.Type) -> V1FeatureScope? {
+        let key = String(describing: T.self)
+
+        guard let feature = v1Features[key] as? V1Feature else {
+            return nil
+        }
+
+        return DatadogCoreFeatureScope(
+            context: v1Context,
+            storage: feature.storage,
+            telemetry: telemetry
+        )
+    }
+
     var context: DatadogV1Context? {
         return v1Context
     }
 }
 
+/// A v1 Feature with an associated stroage.
 internal protocol V1Feature {
+    /// The feature's storage.
     var storage: FeatureStorage { get }
 }
+
+/// This Scope complies with `V1FeatureScope` to provide context and writer to
+/// v1 Features.
+///
+/// The execution block is currently running in `sync`, this will change once the
+/// context is provided on it's own queue.
+internal struct DatadogCoreFeatureScope: V1FeatureScope {
+    let context: DatadogV1Context
+    let storage: FeatureStorage
+    let telemetry: Telemetry?
+
+    func execute(_ block: (DatadogV1Context, Writer) throws -> Void) {
```

Comment:
`execute` by itself doesn't mean much I agree, but as the single method of a protocol called `FeatureScope` it describes what it does: `core.scope(...).execute { }` -> it execute the block in the feature's scope
We could use something more explicit like `executeEventWriteContext` but it looks long and redundant from a usage perspective, no?

The `get` prefix, as proposed in the RFC, should be reserved to method that return something IMO.

> this method cannot (and shouldn't) be used for anything else

Shouldn't we enforce this by design rather than convention? By defining a method that expect an event that is then written to the storage:
```swift
func buildEvent<T>(_ block: (DatadogV1Context) throws -> T?) where T: Encodable
```
the returned event would be then passed to the writer.

Reply:
I see - indeed it makes more sense if thinking of "scope" + "execute" rather than "execute" itself 👍. I'm only worried that we won't stay in sync with Android SDK. One of the goals in the RFC was to embrace a common language, hence we introduced the concept of _"event write context"_ that is a central point to V2 storage architecture. I'm OK with changing it, but we should try to preserve common lingo between both SDKs, as it will make discussion and plannings easier.

> ```swift
> func buildEvent<T>(_ block: (DatadogV1Context) throws -> T?) where T: Encodable
> ```

Mind that soon we will enhance the "event write context" with metadata concept and the `writer` interface will become more specialised:
```swift
// ref. RFC:

internal struct BatchWriter<Event: Codable, Metadata: Codable> {
   let currentMetadata: Metadata
   func write(event: Event, newMetadata: Metadata) { /* ... */ }
}
```
so I don't think that `buildEvent<T>` will be able to support this. I think that starting with convention might be good for the beginning - we can change it later when we understand more. WDYT @maxep?

Reply:
Indeed, I forgot about the metadata :) I will rename the method to `eventWriteContext` to stay aligned with Android for now, we can continue discussing naming as we move forward with the transition.